benchmark-service/benchmark/benchmark.py

```diff
@@ -100,6 +140,29 @@ async def index(request, userdata):  # pylint: disable=unused-argument
     return await render_template('benchmark', request, userdata, 'index.html', context)
 
 
+@router.get('/compare')
+@web_authenticated_developers_only(redirect=False)
+async def compare(request, userdata):  # pylint: disable=unused-argument
+    file1 = request.query.get('file1')
+    file2 = request.query.get('file2')
+    metric = request.query.get('metrics')
+    if file1 is None or file2 is None:
+        benchmarks_context1 = None
```

Comment:
Should it be None, or should this raise an error, or should it only fill in the data for the files it does have?

Reply:
Right now if you don't enter a file name it doesn’t let you press compare saying "please fill out", but if you enter a file name that doesn't exist it gives a page with an error message saying "cannot find file <name>"

Reply:
That sounds good.